wordpress-mobile/WordPress-Android
# Store domains endpoint

This adds storing domains endpoint feature to `DashboardCardDomainSource` and resolves the blinking of the dashboard domain card.

> **Warning**
> Before merging this PR, https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2718 needs to be merged, and fluxc version should be updated here.

**Before**
`DashboardCardDomainSource` was waiting for the response of the fetching domain in order to determine whether to show or hide the domains card.

[before.webm](https://user-images.githubusercontent.com/2471769/233448566-b4561b75-8396-4851-a978-8641a6f1f029.webm)

**After**
Now, `DashboardCardDomainSource` caches the “fetch domains” response, so the previous response can be used instead of waiting for a new "fetch domains" response the next time.

To test:
1. Launch Jetpack app.
2. Select a site with a custom domain.
> **Note**
> If you don't have a site with a custom domain, you can manipulate the code in `DashboardCardDomainUtils` and make
> `hasCustomDomain` always return `false` → `fun hasCustomDomain() = false`
3. Kill the app and relaunch it.
4. Ensure that the dashboard domains card does not appear initially or later.

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Nothing.

5. What automated tests I added (or what prevented me from doing so)
`DashboardCardDomainSourceTest`

PR submission checklist:


UI Changes testing checklist:
